### PR TITLE
BaSP-TG-101: Form - projects

### PR DESCRIPTION
**Pull request by MaxiG-dev**
- Add a redux thunk to GET the entity by its ID
- Implement react-hook-form
- Implement resolver
- Implement Joi validations (see description)
- Implement error messages for validations

### CODE REVIEW

**Review by Ansitox — CHANGES_REQUESTED** [review]
Remember to add a reset button in the form to clean the inputs

**Review by Ansitox — CHANGES_REQUESTED** [review]
The rate inputs are not being validated

**Review by matescarabino — CHANGES_REQUESTED** [review]
The reset button should leave the input Values as they came to the form. If it came through the create it will be blank and if it came through the edit it will be the same data

File: src/Components/Projects/Form/validations.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import Joi from 'joi';
+
+const employeeSchema = Joi.object({
+  employeeId: Joi.string().length(24).required(),
+  role: Joi.string().valid('DEV', 'PM', 'QA', 'TL').required(),
+  rate: Joi.number().required()
+});
+
+export const projectsSchema = Joi.object({
```
Comment (ignacioaristo) [inline src/Components/Projects/Form/validations.js:9]: custom all the messages of the validations.

**Review by ignacioaristo — DISMISSED** [review]

**Review by matescarabino — CHANGES_REQUESTED** [review]
The reset button when editing doesn't bring back the previously deleted employees.

**Review by Ansitox — APPROVED** [review]

**Review by matescarabino — APPROVED** [review]
Approved this week's work. I leave in slack a little thing that would have to fix for the final delivery.

**Review by julianpiaggio — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK


First of all, we wanted to tell you that we consider that your pr complies
the requirements to be approved, and that it can be merged into the master branch 👍

Now, if we consider leaving you some comments:

* When entering when creating and wanting to confirm with the empty fields, all the errors do not jump, but each required individually. Only when entering and putting data in the fields do the validations jump with their corresponding messages. What is expected is that when doing this test and wanting to confirm, all the fields will mark their respective errors at the same time, as in the rest of the dependencies.

projects-week-14


* Having at least two employees in the project and wanting to delete one, always delete the last one. The expected thing would be to delete the option chosen by oneself.

Here we leave you a detailed link with a video that describes what we tell you:
https://imgur.com/a/ueiI5ej

File: src/Components/Projects/Form/validations.js (hunk 1 above)
Comment (MaxiG-dev) [reply src/Components/Projects/Form/validations.js:20]: Done!

**Pull request merged**